homeassistant/components/baf/light.py

```diff
@@ -0,0 +1,105 @@
+"""Support for Big Ass Fans lights."""
+from __future__ import annotations
+
+from typing import Any
+
+from aiobafi6 import Device, OffOnAuto
+
+from homeassistant import config_entries
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_COLOR_TEMP,
+    ColorMode,
+    LightEntity,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.color import (
+    color_temperature_kelvin_to_mired,
+    color_temperature_mired_to_kelvin,
+)
+
+from .const import DOMAIN
+from .entity import BAFEntity
+from .models import BAFData
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up BAF lights."""
+    data: BAFData = hass.data[DOMAIN][entry.entry_id]
+    if data.device.has_light:
+        klass = BAFFanLight if data.device.has_fan else BAFStandaloneLight
+        async_add_entities([klass(data.device)])
+
+
+class BAFLight(BAFEntity, LightEntity):
+    """Representation of a Big Ass Fans light."""
+
+    @callback
+    def _async_update_attrs(self) -> None:
+        """Update attrs from device."""
+        self._attr_is_on = self._device.light_mode == OffOnAuto.ON
+        if self._device.light_brightness_level is not None:
+            self._attr_brightness = int(
+                min(255, self._device.light_brightness_level * 16)
+            )
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the light."""
+        if (brightness := kwargs.get(ATTR_BRIGHTNESS)) is not None:
+            # set the brightness, which will also turn on/off light
+            if brightness == 255:
+                brightness = 256  # this will end up as 16 which is max
+            self._device.light_brightness_level = int(brightness / 16)
```

Comment:
Let's do a correct scaling instead:
```py
            self._device.light_brightness_level = int(brightness / 255 * 16)
```

Or if 0 turns the light off, do like this:
```py
            self._device.light_brightness_level = max(int(brightness / 255 * 16), 1)
```

Reply:
The logic came from the original haiku integration.    Your suggestion is much better.  👍 

0 does seem to turn it of so lets go with the scale + clamp

Reply:
89513734a76437ce82f8ef9e4e956ac360d43176